### PR TITLE
Bugfix/ab#69 99 103 105 data validation

### PR DESCRIPTION
**Pull request by BerendWouters**

### CODE REVIEW

File: Singer.API/ClientApp/src/app/modules/core/utils/custom-date-validators.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   return (control: AbstractControl): { [key: string]: any } | null => {
+      const formDate = new Date(control.value);
+      return formDate < date ? { dateNotBefore: { value: formDate } } : null;
+   };
+}
+
+export function dateNotAfter(date: Date): ValidatorFn {
+   return (control: AbstractControl): { [key: string]: any } | null => {
+      const formDate = new Date(control.value);
+      return formDate > date ? { dateNotBefore: { value: formDate } } : null;
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/core/utils/custom-date-validators.ts]: `dateNotBefore` should be `dateNotAfer`

File: Singer.API/ClientApp/src/app/modules/core/utils/custom-date-validators.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   };
+}
+
+export function dateBetween(first: Date, last: Date) {
+   return (control: AbstractControl): { [key: string]: any } | null => {
+      const formDate = new Date(control.value);
+      return formDate < first || formDate > last
+         ? { dateNotBefore: { value: formDate } }
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/core/utils/custom-date-validators.ts]: `dateNotBefore` should be `dateBetween`

File: Singer.API/Helpers/Exceptions/BadInputException.cs [hunk 3]
```diff
@@ -2,49 +2,41 @@
 
 namespace Singer.Helpers.Exceptions
 {
-   /// <summary>
-   /// Exception that represents a 400 http status code.
-   /// </summary>
+   /// <summary>Exception that represents a 400 http status code.</summary>
```
Comment (BerendWouters) [inline Singer.API/Helpers/Exceptions/BadInputException.cs:5]: Faulty merge.

**Review by BerendWouters — COMMENTED** [review]
We should discuss name length requirements

File: Singer.API/Helpers/Exceptions/BadInputException.cs (hunk 3 above)
Comment (wim07101993) [reply Singer.API/Helpers/Exceptions/BadInputException.cs:5]: how is this a faulty merge?

Comment (wim07101993) [reply Singer.API/Helpers/Exceptions/BadInputException.cs:5]: If the message should change, what should it become?

**Review by BerendWouters — COMMENTED** [review]
Good fixes

File: Singer.API/Helpers/Exceptions/BadInputException.cs (hunk 3 above)
Comment (BerendWouters) [reply Singer.API/Helpers/Exceptions/BadInputException.cs:5]: Message should be `Er is iets misgegaan.` (or alike)

**Review by BerendWouters — COMMENTED** [review]
Since I made this PR, I cannot merge it.

File: Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-details/singerevent-details.component.html [hunk 4]
```diff
@@ -110,11 +136,32 @@ <h3 class="section-title">Evenement</h3>
...
+               >
+                  Aantal toegelaten personene is een verplicht veld.
+               </div>
+               <div
+                  *ngIf="
+                     formControlGroup.controls.maxRegistrantsFieldControl.errors
+                        .min
+                  "
+               >
+                  Aantal toegelaten personene mag minstens
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-details/singerevent-details.component.html]: `mag` should be `moet`

Comment (BerendWouters) [reply Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-details/singerevent-details.component.html]: Also typo in `personene`

File: Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-details/singerevent-details.component.html [hunk 5]
```diff
@@ -148,6 +207,7 @@ <h3 class="section-title">Datums</h3>
          <mat-form-field>
             <input
                matInput
+               [min]="today"
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-details/singerevent-details.component.html:210]: Makes sense

**Review by BerendWouters — COMMENTED** [review]
Some typos remain

**Review by BerendWouters — COMMENTED** [review]
I approve of this.

**Review by vyruz1986 — APPROVED** [review]
Proxy approval :)

**Pull request merged**